Source/CompSpinalEngineTrail.cs

```diff
@@ -0,0 +1,179 @@
+﻿using RimWorld;
+using SaveOurShip2;
+using System.Text;
+using Verse;
+
+namespace TheCafFiend
+{
+    [StaticConstructorOnStartup]
+    public class CompSpinalEngineTrail : CompEngineTrail
+    {
+        public override string CompInspectStringExtra()
+        {
+            StringBuilder stringBuilder = new StringBuilder();
+            
+            if (fullyFormed)
+            {
+                stringBuilder.Append($"Fuel burn rate per second: {cachedFuelUse} of {cachedFuelAllowed} supported");
+                stringBuilder.AppendInNewLine($"Thrust: {cachedThrust * 500}");
+            }
+            else
+            {
+                stringBuilder.Append(cachedError.Colorize(UnityEngine.Color.red));
+            }
+            return stringBuilder.ToString();
+        }
+        public virtual CompProperties_SpinalEngineTrail Properties //Wait is this dirty I just want to be clear it's not hiding hrm
+        {
+            get { return props as CompProperties_SpinalEngineTrail; }
+        }
+        public class ReturnValue
```

Comment:
Basically I needed to return an entire bundle of values since I was iterating over a multiblock and each bit could alter multiple values: A class to drop back seemed? like the best way but I am super open to a different answer to this. It's very literally values, to be returned, from the recalc

Reply:
So anything you return from a function is a return value right?  Six months from now you'll probably want something like "EngineState" to come out of recalc, since that's what you're effectively representing here.

Reply:
OH what *specifically* does this represent vs what is the basic type. Yeah 100%